src/__tests__/type.js

```diff
@@ -715,3 +715,77 @@ test('should give error if we are trying to call type on an invalid element', as
     `"the current element is of type BODY and doesn't have a valid value"`,
   )
 })
+
+test('navigation key: {arrowleft} and {arrowright} moves the cursor', () => {
+  const {element, getEventSnapshot} = setup('<input />')
+  userEvent.type(element, 'Brea{arrowleft}k{arrowright}fast')
+  expect(getEventSnapshot()).toMatchInlineSnapshot(`
+    Events fired on: input[value="Brekafast"]
+
+    input[value=""] - pointerover
+    input[value=""] - pointerenter
+    input[value=""] - mouseover: Left (0)
+    input[value=""] - mouseenter: Left (0)
+    input[value=""] - pointermove
+    input[value=""] - mousemove: Left (0)
+    input[value=""] - pointerdown
+    input[value=""] - mousedown: Left (0)
+    input[value=""] - focus
+    input[value=""] - focusin
+    input[value=""] - pointerup
+    input[value=""] - mouseup: Left (0)
+    input[value=""] - click: Left (0)
+    input[value=""] - keydown: B (66)
+    input[value=""] - keypress: B (66)
+    input[value="B"] - input
+      "{CURSOR}" -> "B{CURSOR}"
+    input[value="B"] - keyup: B (66)
+    input[value="B"] - keydown: r (114)
+    input[value="B"] - keypress: r (114)
+    input[value="Br"] - input
+      "B{CURSOR}" -> "Br{CURSOR}"
+    input[value="Br"] - keyup: r (114)
+    input[value="Br"] - keydown: e (101)
+    input[value="Br"] - keypress: e (101)
+    input[value="Bre"] - input
+      "Br{CURSOR}" -> "Bre{CURSOR}"
+    input[value="Bre"] - keyup: e (101)
+    input[value="Bre"] - keydown: a (97)
+    input[value="Bre"] - keypress: a (97)
+    input[value="Brea"] - input
+      "Bre{CURSOR}" -> "Brea{CURSOR}"
+    input[value="Brea"] - keyup: a (97)
+    input[value="Brea"] - keydown: ArrowLeft (37)
+    input[value="Brea"] - select
+    input[value="Brea"] - keyup: ArrowLeft (37)
+    input[value="Brea"] - keydown: k (107)
+    input[value="Brea"] - keypress: k (107)
+    input[value="Breka"] - input
+      "Bre{CURSOR}a" -> "Breka{CURSOR}"
```

Comment:
I tried looking into this but got a bit lost. It looks like the event snapshot has a bug where the `{CURSOR}` is always at the end. I think to get an accurate representation of the navigation keys in our tests then the event snapshot should move the cursor around as well. 

Any advice or tips on getting this fixed?

Reply:
I think I'm figuring out how the test utils work. 

After learning a bit more, I think I'm going to implement a change for arrows (?):

```
    input[value="Brea"] - keydown: ArrowLeft (37)
    input[value="Brea"] - select
      "Brea{CURSOR}" -> "Bre{CURSOR}a"
    input[value="Brea"] - keyup: ArrowLeft (37)
```

Though, I'm not sure if it should be on the select event or the keyup.

Reply:
I'm not sure. Based on [the playground](https://codesandbox.io/s/user-event-playground-eo909?file=/src/index.js) we shouldn't fire a `select` event at all I think (though it may be unfortunately unavoidable).

Reply:
Looks like it is a bug in jsdom's `HTMLInputElement-impl.js`:

```
set value(val) {
    switch (valueAttributeMode(this.type)) {
      // https://html.spec.whatwg.org/multipage/input.html#dom-input-value-value
      case "value": {
        const oldValue = this._value;
        this._value = sanitizeValueByType(this, val);
        this._dirtyValue = true;

        if (oldValue !== this._value) {
          this._selectionStart = this._selectionEnd = this._getValueLength();
          this._selectionDirection = "none";
        }
        break;
      }

```

where `val = "Breka"` and `oldValue = "Brea"`

Reply:
Okay, nevermind. This is an issue with the HTML standard. See more information @ https://github.com/whatwg/html/issues/5743

I am going to resolve this and ignore the odd cursor position.